### PR TITLE
Fix wazuh install in unify-unattended script

### PR DESCRIPTION
**Pull request by lastranikos**
|Related issue|
|---|
| #1036 |

## Description

- The script tries to install the stable release, even when unstable is chosen. (Ubuntu 20.04)
- The script, when run from outside the storage folder. When trying to install filebeat not finding a config.

### Issue: release

**Install log: wazuh and filebeat**
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --wazuh-server -wn wazuh1 --dev
12/13/2021 15:44:06 INFO: Starting the installation...
12/13/2021 15:44:06 INFO: Installing all necessary utilities for the installation...
12/13/2021 15:44:10 INFO: Done
12/13/2021 15:44:10 INFO: Adding the Wazuh repository...
12/13/2021 15:44:15 INFO: Done
12/13/2021 15:44:15 INFO: Installing the Wazuh manager...
12/13/2021 15:44:16 ERROR: Wazuh installation failed
12/13/2021 15:44:16 WARNING: Cleaning the installation
12/13/2021 15:44:16 WARNING: Installation cleaned. Check the /var/log/wazuh-unattended-installation.log file to learn more about the issue.
```

**Install log: elasticsearch and kibana**
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --kibana --elasticsearch --create-certificates -en node1 --create-certificates --dev
12/13/2021 19:23:10 INFO: Configuration file found. Creating certificates...
12/13/2021 19:23:10 INFO: Creating the Elasticsearch certificates...
12/13/2021 19:23:10 INFO: Creating Wazuh server certificates...
12/13/2021 19:23:10 INFO: Creating Kibana certificate...
12/13/2021 19:23:11 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/certs.
12/13/2021 19:23:11 INFO: Starting the installation...
12/13/2021 19:23:11 INFO: Installing all necessary utilities for the installation...
12/13/2021 19:23:46 INFO: Done
12/13/2021 19:23:46 INFO: Adding the Wazuh repository...
12/13/2021 19:23:51 INFO: Done
head: cannot open './config.yml' for reading: No such file or directory
12/13/2021 19:23:51 INFO: Installing Open Distro for Elasticsearch...
12/13/2021 19:29:47 INFO: Done
12/13/2021 19:29:47 INFO: Configuring Elasticsearch...
head: cannot open './config.yml' for reading: No such file or directory
awk: fatal: cannot open file `./config.yml' for reading (No such file or directory)
head: cannot open './config.yml' for reading: No such file or directory
12/13/2021 19:29:49 INFO: Elasticsearch installed.
12/13/2021 19:29:49 INFO: Starting Elasticsearch...
12/13/2021 19:30:06 INFO: Elasticsearch started
12/13/2021 19:30:06 INFO: Initializing Elasticsearch...
```

**debug with 'set -x'**
```
+ eval 'echo deb' '${repobaseurl}/apt/' stable 'main | tee /etc/apt/sources.list.d/wazuh.list > /dev/null 2>&1'
++ echo deb https://packages-dev.wazuh.com/pre-release/apt/ stable main
++ tee /etc/apt/sources.list.d/wazuh.list
```

**debug: apt update**
```
Ign:5 https://packages-dev.wazuh.com/pre-release/apt stable InRelease
Err:6 https://packages-dev.wazuh.com/pre-release/apt stable Release
  404  Not Found [IP: 13.227.92.21 443]
Reading package lists...
E: The repository 'https://packages-dev.wazuh.com/pre-release/apt stable Release' does not have a Release file.
```

### Issue: filebeat

```
12/13/2021 16:43:38 INFO: Installing Filebeat...
awk: fatal: cannot open file `./config.yml' for reading (No such file or directory)
awk: fatal: cannot open file `./config.yml' for reading (No such file or directory)
12/13/2021 16:44:07 INFO: Done
```

## Tests remediation filebeat and release:
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --wazuh-server -wn wazuh1 --dev
12/13/2021 17:53:05 INFO: Starting the installation...
12/13/2021 17:53:05 INFO: Installing all necessary utilities for the installation...
12/13/2021 17:53:41 INFO: Done
12/13/2021 17:53:41 INFO: Adding the Wazuh repository...
12/13/2021 17:53:46 INFO: Done
12/13/2021 17:53:46 INFO: Installing the Wazuh manager...
12/13/2021 17:56:09 INFO: Done
12/13/2021 17:56:30 INFO: Wazuh-manager started
12/13/2021 17:56:30 INFO: Installing Filebeat...
12/13/2021 17:56:59 INFO: Done
12/13/2021 17:56:59 INFO: Starting Filebeat...
12/13/2021 17:57:00 INFO: Filebeat started
12/13/2021 17:57:00 INFO: Setting the Wazuh repository to production
12/13/2021 17:57:00 INFO: Done
```

## Tests remediation elasticsearch and kibana:
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --kibana --elasticsearch --create-certificates -en node1 --create-certificates --dev


12/13/2021 19:55:26 INFO: Configuration file found. Creating certificates...
12/13/2021 19:55:26 INFO: Creating the Elasticsearch certificates...
12/13/2021 19:55:26 INFO: Creating Wazuh server certificates...
12/13/2021 19:55:26 INFO: Creating Kibana certificate...
12/13/2021 19:55:26 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/certs.
12/13/2021 19:55:26 INFO: Starting the installation...
12/13/2021 19:55:26 INFO: Installing all necessary utilities for the installation...
12/13/2021 19:56:06 INFO: Done
12/13/2021 19:56:06 INFO: Adding the Wazuh repository...
12/13/2021 19:56:11 INFO: Done
12/13/2021 19:56:11 INFO: Installing Open Distro for Elasticsearch...
12/13/2021 20:02:50 INFO: Done
12/13/2021 20:02:50 INFO: Configuring Elasticsearch...
12/13/2021 20:02:51 INFO: Elasticsearch installed.
12/13/2021 20:02:51 INFO: Starting Elasticsearch...
12/13/2021 20:03:07 INFO: Elasticsearch started
12/13/2021 20:03:07 INFO: Initializing Elasticsearch...

12/13/2021 20:03:07 INFO: Done
12/13/2021 20:03:07 INFO: Done
12/13/2021 20:03:07 INFO: Setting the Wazuh repository to production
12/13/2021 20:03:07 INFO: Done
12/13/2021 20:03:07 INFO: Starting the installation...
12/13/2021 20:03:07 INFO: Installing all necessary utilities for the installation...
12/13/2021 20:03:12 INFO: Done
12/13/2021 20:03:12 INFO: Adding the Wazuh repository...
12/13/2021 20:03:12 INFO: Wazuh repository already exists skipping
12/13/2021 20:03:12 INFO: Done
12/13/2021 20:03:12 INFO: Installing Open Distro for Kibana...
12/13/2021 20:03:13 ERROR: Kibana installation failed
12/13/2021 20:03:13 WARNING: Cleaning the installation
12/13/2021 20:03:13 WARNING: Removing Elasticsearch...
12/13/2021 20:03:20 WARNING: Installation cleaned. Check the /var/log/wazuh-unattended-installation.log file to learn more about the issue.
```

## Tests remediation Prerequisites: case with error and exit

```
pwd
/tmp/unattended_scripts

rm /tmp/unattended_scripts/config.yml 
rm: cannot remove '/tmp/unattended_scripts/config.yml': No such file or directory

bash /tmp/unattended_scripts/wazuh_install.sh --all-in-one --local --ignore-health-check --dev
12/14/2021 21:48:25 WARNING: The file /tmp/unattended_scripts/config.yml is required. It should be located in the script directory:/tmp/unattended_scripts
```

## Tests remediation Prerequisites: normal case

```
bash /tmp/unattended_scripts/wazuh_install.sh --local --wazuh-server -wn wazuh1 --dev
12/14/2021 21:51:46 INFO: checkPrerequisites
12/14/2021 21:51:46 INFO: Starting the installation...
12/14/2021 21:51:46 INFO: Installing all necessary utilities for the installation...
```

### CODE REVIEW

File: unattended_scripts/install_functions/opendistro/filebeat.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ configureFilebeat() {
         echo "  - ${nip}"  >> /etc/filebeat/filebeat.yml
     else
         echo "output.elasticsearch.hosts:" >> /etc/filebeat/filebeat.yml
-        sh=$(awk -v RS='' '/discovery.seed_hosts:/' ./config.yml)
+        sh=$(awk -v RS='' '/discovery.seed_hosts:/' ${base_path}/config.yml)
```
Comment (rauldpm) [inline unattended_scripts/install_functions/opendistro/filebeat.sh:48]: Change this in kibana.yml, common.yml and elasticsearch.yml too. When using files from S3, when is this file downloaded?

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, minor changes

File: unattended_scripts/install_functions/opendistro/filebeat.sh (hunk 1 above)
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/filebeat.sh:48]: Done

**Review by verdx — APPROVED** [review]
LGTM

File: unattended_scripts/wazuh_install.sh [hunk 2]
```diff
@@ -286,22 +282,24 @@ main() {
...
         installElasticsearch
         configureElasticsearchAIO
         installFilebeat
         configureFilebeatAIO
         installKibana
         configureKibanaAIO
+    fi
+
+    if [ -n "${AIO}" ] || [ -n "${wazuh}" ] || [ -n "${kibana}" ] || [ -n "${elastic}" ]; then
+        logger -w "restoreWazuhrepo"
```
Comment (verdx) [inline unattended_scripts/wazuh_install.sh:299]: Not a very explanatory comment. A better option would be "Restoring Wazuh repository"

**Review by verdx — CHANGES_REQUESTED** [review]

File: unattended_scripts/wazuh_install.sh (hunk 2 above)
Comment (lastranikos) [reply unattended_scripts/wazuh_install.sh]: Done

**Review by verdx — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**